app/models/manageiq/providers/autosde/inventory/collector/target_collection.rb

```diff
@@ -1,8 +1,9 @@
 class ManageIQ::Providers::Autosde::Inventory::Collector::TargetCollection < ManageIQ::Providers::Autosde::Inventory::Collector::StorageManager
-  def initialize(_manager, _target)
+  def initialize(_manager, target)
     super
 
     parse_targets!
+    target.manager_refs_by_association_reset
```

Comment:
NOTE this is needed because the `manager_refs_by_association` is cached when initially building the inventory collections, so later calls to `#add_target` don't update this.  I think this should be handled by the inventory_refresh gem when you call add_target so I'll work on adding that.